src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -336,6 +338,7 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
     internal var superclass: TypeName = ANY
     internal val superclassConstructorParameters = mutableListOf<CodeBlock>()
     internal val superinterfaces = mutableListOf<TypeName>()
+    internal val superinterfaceDelegates = mutableListOf<ParameterSpec>()
```

Comment:
What if we had a `superinterfacesWithDelegates = mutableMapOf<TypeName, String>()` instead?